# ML-02 — Research Question and Provisional Lane

[![Open In Colab](https://colab.research.google.com/assets/colab-badge.svg)](https://colab.research.google.com/github/Farida-Hany7/flyrank/blob/main/work/notebooks/w01_research_question.ipynb)

This skeleton is yours to fill. Work the sections **in order** — each one has a one-line hint. Simple words, honest numbers.

> Working with an AI assistant? Tell it to read `skills/README.md` first and load the one skill this assignment names on its card.

In [16]:
!git clone https://github.com/Farida-Hany7/flyrank.git

fatal: destination path 'flyrank' already exists and is not an empty directory.


In [17]:
import pandas as pd

df = pd.read_csv("/content/flyrank/data/raw/content_refresh_anonymized.csv")
print(df.shape)

(30000, 44)


In [18]:
df.columns.tolist()


['content_id',
 'client_id',
 'search_volume',
 'competition',
 'competition_level',
 'cpc',
 'content_type',
 'main_intent',
 'word_count',
 'char_count',
 'provider_used',
 'model_used',
 'impressions_90d',
 'clicks_90d',
 'pageviews_90d',
 'sessions_90d',
 'users_90d',
 'engaged_sessions_90d',
 'ai_sessions_90d',
 'scroll_events_90d',
 'days_with_impressions',
 'days_with_sessions',
 'impressions_last_30d',
 'clicks_last_30d',
 'sessions_last_30d',
 'impressions_prev_30d',
 'clicks_prev_30d',
 'sessions_prev_30d',
 'content_age_days',
 'age_tier',
 'age_tier_order',
 'days_since_last_update',
 'freshness_tier',
 'word_count_tier',
 'char_count_tier',
 'ctr',
 'avg_position',
 'engagement_rate',
 'scroll_rate',
 'ai_traffic_pct',
 'impression_tier',
 'position_tier',
 'trend_direction',
 'trend_pct']

## 1. My lane (or freestyle) and why

*Name your lane — or say 'freestyle' and describe your own question. One short paragraph: why this one?*

**Lane: Content decline prioritization**

I chose this lane because editors may have many content items but limited time to review them. I want to study which observed signals can help identify content that may need attention first.


In [19]:
print("Rows:", len(df))
print("Clients:", df["client_id"].nunique())
print("\nTrend direction counts:")
print(df["trend_direction"].value_counts(dropna=False))

print("\nMedian trend_pct:", df["trend_pct"].median())
print("Median days since last update:", df["days_since_last_update"].median())

Rows: 30000
Clients: 32

Trend direction counts:
trend_direction
down      16262
stable     5962
up         4388
new        2236
flat       1152
Name: count, dtype: int64

Median trend_pct: -33.5
Median days since last update: 20.0


## 2. The question: decision, action, cost of a wrong call

*What decision does your work improve? Who acts on it? What does a wrong recommendation cost?*

**Research question:** Which observed signals can help prioritize content for editorial review?

An editor would use the results to decide what to review first. A wrong recommendation could waste time or cause declining content to be missed. The goal is to support the editor’s decision, not make it automatically.


## 3. Quick look at the data (2-3 real numbers)

*Load the starter CSV below and show 2-3 real numbers that make your lane look worth the next 7 weeks.*



The dataset has 30,000 content items from 32 clients. 16,262 items (about 54%) have an observed downward trend, and the median trend_pct is -33.5. This makes content decline worth investigating.

## 4. Careful words: what I can and can't claim

*Write what your work will be able to say (observed, directional, decision-support) — and what it never will (causal proof, 'predicting Google').*



This work can show observed patterns and directional signals that may support editorial decisions. It cannot prove causation or claim to predict Google rankings or future performance with certainty.

## Self-check

Before you submit, confirm each line honestly:

- - [x] Every section above is filled — markdown thinking AND the code that backs it
- - [x] The notebook runs top to bottom with no errors (Runtime → Run all)
- - [x] No client names, URLs, or private queries anywhere
- - [x] My claims use careful words: observed, measured, directional, decision-support
- - [x] Committed to my repo under `work/notebooks/` — then submit your repo URL on the card. Done.